STOR-1726: auto-generate assets for CIFS/Samba driver

mpatlasov opened the pull request:
The patch ensures that all assets in `overlays/samba/generated/standalone` are generated by `generator`. Resulting assets survive simple dynamic-provisioning test with `test/e2e/samba/samba-server.yaml`

jsafrane commented:
I can see smb-operator-e2e did not pass, probably a typo somewhere around livenessprobe.

mpatlasov commented:
/test smb-operator-e2e

mpatlasov commented:
Last force-push rebased local branch to the head of origin/master (commit 097858be55)

mpatlasov commented:
Last force-push fixed a typo in `node_add_driver.yaml` patch.

mpatlasov commented:
Last force-push fixed a typo in `controller_add_driver.yaml` patch.

jsafrane commented on cmd/generator/main.go:
```diff
@@ -23,7 +24,14 @@ func main() {
 	cfgs := collectConfigs()
 	for _, cfg := range cfgs {
-		for _, flavour := range []generator.ClusterFlavour{generator.FlavourStandalone, generator.FlavourHyperShift} {
+		var flavour_range []generator.ClusterFlavour
```
nit: Go does not use snake_case. IMO simple `flavours` would do

jsafrane commented on assets/overlays/samba/generated/standalone/service.yaml:
```diff
@@ -0,0 +1,31 @@
+spec:
+  ports:
+  - name: provisioner-m
+    port: 9222
```
This is 9223 in controller.yaml

mpatlasov replied:
This is general problem - a bug. I'm fixing it in another commit:

`ad038fd8f80b111bee711ac1e813ec9d253d65ae` STOR-1726: fix port calculation in the generator

Surprisingly, our CI jobs have never caught it for aws-ebs / azure-disk / azure-file :(

mpatlasov commented:
/test e2e-azure-ovn-upgrade

mpatlasov commented:
Last force-push:
 * fixed `cmd/generator/main.go`: s/flavour_range/flavours`
 * fixed mismatch between port number in `service.yaml` and `controller.yaml`: [new commit](https://github.com/openshift/csi-operator/pull/217/commits/ad038fd8f80b111bee711ac1e813ec9d253d65ae)

mpatlasov commented:
/test e2e-azure-ovn-upgrade

jsafrane commented:
/lgtm
/retest-required

mpatlasov commented:
/test e2e-azure-ovn-upgrade

mpatlasov commented:
/test e2e-azure-ovn-upgrade

mpatlasov commented:
/test e2e-azure-ovn-upgrade

mpatlasov commented:
/test e2e-azure-ovn-upgrade

jsafrane commented:
/label docs-approved
/label px-approved
/label qe-approved

We do post-merge tests, when it gets to OLM index image.
